accounts/ethstore/src/import.rs

```diff
@@ -46,21 +46,22 @@ pub fn import_accounts(src: &dyn KeyDirectory, dst: &dyn KeyDirectory) -> Result
 		.map(|a| a.address)
 		.collect::<HashSet<_>>();
 
-	accounts.into_iter()
-		.filter(|a| !existing_accounts.contains(&a.address))
-		.map(|a| {
-			let address = a.address.clone();
-			dst.insert(a)?;
-			Ok(address)
-		}).collect()
+	let mut out = Vec::with_capacity(accounts.len());
```

Comment:
I guess that technically the lint triggered this was `map().filter()` instead `map_filter()`?

Then you switch to a for loop instead because it was not possible to return the error in `filter_map`?

Reply:
Yes, I simply didn't want to untangle the mix of `Option` and `Result`. And in general the imperative approach seems to me much clearer than the map-and-collect of results. It also returns earlier (on first error) although I'm not sure if it's the desired behavior here.

Reply:
right, it seems that the actual implementation of `KeyDirectory::insert` always returns `Ok` but I don't know, let's ignore it for this PR.